app/models/relationship.rb

```diff
@@ -205,6 +205,23 @@ def self.create_unless_exists(source_id, target_id, relationship_type, options =
     r
   end
 
+  def self.replicate_status_to_children(pm_id, uid, tid)
+    pm = ProjectMedia.find_by_id(pm_id)
+    return if pm.nil?
+    User.current = User.where(id: uid).last
+    Team.current = Team.where(id: tid).last
+    status = pm.last_verification_status
+    pm.source_relationships.confirmed.find_each do |relationship|
+      s = relationship.target.last_status_obj
+      next if s.nil? || s.status == status
+      s.status = status
+      s.bypass_status_publish_check = true
+      s.save!
```

Comment:
Sawy, I think this part can be refactored as its own instance method for the `Relationship` model, something like `sync_statuses`.